Convert hui-input-text-entity-row to TypeScript/LitElement

iantrich opened the pull request.

iantrich committed:
Convert hui-input-text-entity-row to TypeScript/LitElement

zsarnett commented on src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts:
```diff
@@ -0,0 +1,90 @@
+
+import { HomeAssistant } from "../../../types";
+import { EntityRow, EntityConfig } from "./types";
+import { HassEntity } from "home-assistant-js-websocket";
```
Absolute Import above Relative

iantrich committed:
Address review comments

balloob commented on src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts:
```diff
@@ -0,0 +1,90 @@
+    }
+
+    return html`
+      <hui-generic-entity-row .hass="${this.hass}" .config="${this._config}">
```
Prettier will make it 1 line if it's small. It's fine, we use prettier to not have to have this discussion :D

balloob commented on src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts:
```diff
@@ -0,0 +1,90 @@
+      return;
+    }
+
+    this.hass!.callService("input_text", "set_value", {
```
I think that we should extract this to `data/input_text.ts`

balloob commented on src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts:
```diff
@@ -0,0 +1,90 @@
+      return html``;
+    }
+
+    this._stateObj = this.hass.states[this._config.entity];
```
No reason for this to be an instance variable, is it? Can just be a local one?

balloob commented on src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts:
```diff
@@ -0,0 +1,90 @@
+  }
+
+  private get _value(): PaperInputElement {
+    return this.shadowRoot!.querySelector(".value") as PaperInputElement;
```
No need to add a class, you could just query for `paper-input` as there is only one in this shadow DOM.

The nice thing about ShadowDOM is that we have a mini piece of DOM that is only whatever we put in our template. No need to wiggle around with class selectors etc.

balloob commented on src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts:
```diff
@@ -0,0 +1,90 @@
+    `;
+  }
+
+  private get _value(): PaperInputElement {
```
This name is a bit weird, as you are returning an input element, not a value. I would call it `inputEl`

iantrich committed:
Address review comments

Does anyone know why line 74 seemingly doesn't fire when the value hasn't changed? I think we should blur regardless if the value changed or not on "enter" but only seems to work when setValue is called

balloob committed:
Return promise from call service
